### DISCUSSION TITLE
Cannot get Formatted MDX to show for Blog Posts. Getting ERRORs now.

### DISCUSSION THREAD

**Discussion by davidkartuzinski** [gatsbyjs/gatsby#36519] [Help]
I am updating to Gatsby v4 from a theme I coded three years ago.

Following the new tutorial I was able to create new pages using `{mdx.frontmatter__slug.js}` and `<MDXProvider>` with almost no issues. [`{children}` never worked for me without using `<MDXProvider>`..., but it eventually worked great!]

[rendering worked like this.](https://github.com/davidkartuzinski/gatsy-theme-naked-core-2022/blob/85cc22f50fe6dd1192e20864a7d3fb551f704cc3/src/pages/blog/%7Bmdx.frontmatter__slug%7D.js#L79) on my **`MAIN`** branch.

However, using `{mdx.frontmatter__slug.js}` did not allow for me to create **PREVIOUS/NEXT** blog page navigation like [this](https://toripugh.com/blog/gatsby-blog--next-and-previous-links) as I did before [here](https://github.com/davidkartuzinski/gatsby-theme-naked-core/blob/master/src/templates/single-blog-post.js) 3 years ago with `gatsby-node.js` and `createPages`.

Using `gatsby-node.js` and `createPages` everything is working as expected **EXCEPT** the `body` of my MDX blog posts. They are rendered **_unformatted_** if I use `data.mdx.body` and `{children}` doesn't work with either `<MDXRenderer>` or `<MDXProvider>` or by itself.

Under my **`Develop`** branch, [the unworkable code](https://github.com/davidkartuzinski/gatsy-theme-naked-core-2022/blob/19daa53f014bf7f612a36159f412d9f60bd98086/src/templates/single-blog-post.js#L88). 

I have tried everything, but using `<MDXRenderer>` or `<MDXProvider>` - I get the following error message:

```
Unhandled Runtime Error
Close
One unhandled runtime error found in your files. See the list below to fix it:


Error in function createFiberFromTypeAndProps in ./node_modules/react-dom/cjs/react-dom.development.js:28389
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports. Check the render method of `BlogPost`.
```
and
```
./node_modules/react-dom/cjs/react-dom.development.js:28389

  28387 |           }
  28388 |
> 28389 |           throw new Error('Element type is invalid: expected a string (for built-in ' + 'components) or a class/function (for composite components) ' + ("but got: " + (type == null ? type : typeof type) + "." + info));
  28390 |         }
  28391 |     }
  28392 |   }
```
I have searched everywhere. 

from the docs [Using Page templates](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#updating-page-templates), whenever I try to destructure like this referring to `data`, the example, `function PostTemplate({ data: { mdx }, children })`. I get errors saying `data is not defined...`. And the rest of the example using just `{children}` does not work for me. `function PostTemplate({ data, children })` doesn't work for me.

from the tutorial [here on Part 6](https://www.gatsbyjs.com/docs/tutorial/part-6/#render-post-contents-in-the-blog-post-page-template) this never worked for me. I had to use `<MDXProvider>`.

Even @LekoArts great blog, [prev/next tutorial](https://www.lekoarts.de/gatsby/tips-and-tricks-for-gatsby#previousnext---links) - does not work for me with formatted markdown.

Please help. I have looked into every theme and there doesn't seem to be any one of them employing Gatsby v4 with `{children}`, `{mdx.frontmatter__slug.js}`, and **PREV/NEXT**.

Thank you.

**Comment by ovflowd** [comment]
Same here, Gatsby Plugin MDX v4 is unusable. Can't make it work at all, it doesn't matter how hard I try.

This is quite stupid as there are no examples out there using the v4 of the plugin.

**Reply by ovflowd** [reply]
In my case, the plugin is working neither by sourcing the pages on (https://www.gatsbyjs.com/docs/how-to/routing/mdx/#create-pages-from-sourced-mdx-files) nor by using dynamic layouts (https://www.gatsbyjs.com/docs/how-to/routing/mdx#make-a-layout-template-for-your-posts).

This is truly frustrating as it gives a variety of different errors. For example with TypeScript and dynamic layouts it accuses that `interface` or `type` are unknown symbols (syntax error).

And by sourcing the MDX files directly, it says that the `---` (Frontmatter) is an unallowed left-padded assignment. This is frustrating as not only is the v4 a breaking change, but it is a breaking change in every possible manner, which requires us (the maintainers of our pages) to switch and refactor everything completely. Not to mention simple features such as i18n become straight out of consideration, as the new version doesn't even allow you to query the content directly from pages.

I wonder if the maintainers at least tried to take certain cases in consideration...

**Reply by davidkartuzinski** [reply]
Hello. I finally got something to work. You can see everything that I did here. https://github.com/davidkartuzinski/gatsy-theme-naked-core - I just could not use the new dynamic link structure. I think  I gave up and just used the old school patterns until all the plugins are updated. I am use MDX v 4 with the gatsby plugin.

**Reply by LekoArts** [reply]
@ovflowd 

When we released the new major we also released https://github.com/gatsbyjs/gatsby/tree/master/examples/using-mdx, so there is an update to date example. Also one where I tried to show as many variations and things from the README as possible.

As we wrote in https://www.gatsbyjs.com/docs/reference/release-notes/v4.21/#gatsby-plugin-mdx-v4 it's obviously a new major version and we had to do some bigger breaking changes for the sake of making maintenance of new versions _way_ easier. I explicitly added that note as I understand that it's a bigger migration than previous major bumps. So I don't understand where the complaint is coming from that a new major version includes breaking changes?

I'm happy to help with specific questions (with code examples, repos) and @PaulieScanlon is happy to help, too, but sadly I can't be of help with vague "it doesn't work" complaints. We tried our best to dogfood the new version on all of our projects (where it worked) but we can't cover every use case, it's just not possible. But we're happy to review those when they trickle in and make adjustments (code fixes, docs updates, etc.) as needed.

**Reply by PaulieScanlon** [reply]
Hey @ovflowd the error you're seeing that points towards the frontmatter. 

Is it similar to this: 

"Invalid left-hand side in prefix operation"

If so, I've detailed what this is and a few ways to find and replace the problem characters that are breaking changes in MDX 2

https://paulie.dev/posts/2022/08/mdx-2-breaking-changes-and-gatsby-plugin-mdx-v4/

**Reply by ovflowd** [reply]
Thank you all, @LekoArts and @PaulieScanlon, for your replies. I truly appreciate taking the time to get an 👀 in my use case. I apologise for the quite stressed-out comment beforehand; I was exhausted after multiple days of trying to make this happen.

I believe my use case is quite the thing, as it's an entirely non-conventional scenario. I truly believe migrating to MDX v2 itself will be a task that is quite complex and very impossible due to time limitations.

To start, our Markdown files take a very unconventional approach. Some completely disobey a few minor MDX v2 syntax rules that are mandatory and straight, causing MDX v2 compilation to fail. For example, `<link to somewhere>` instead of `[link](to somewhere)`. We must mention that we have hundreds of Markdown files, some thousands of lines long. (This is an example of the original Markdown file: https://github.com/nodejs/node/blob/main/doc/api/fs.md, and here (https://raw.githubusercontent.com/nodejs/nodejs.dev/main/content/api/v18/fs.en.md) it is after it is parsed by a script we created that already makes it MDX compatible and sanitises as much as possible the content.

Another road blocker is that our website uses `gatsby-theme-i18n` (modified), and most of our pages have queries to retrieve the localised markdown file for that language or fallback to the default language (https://github.com/nodejs/nodejs.dev/blob/main/src/pages/community.tsx#L11) this is done because sadly Gatsby's GraphQL schema doesn't support **`OR`** operators. Which is a shame.

The new plugin states that no named queries can be done, so I have no idea how this would be accomplished; even if the query worked, how would the data be transmitted to "children" prop?

Next, another road blocker was that, for some reason, any TypeScript layout receives the error `Parsing error: The keyword 'interface' is reserved` coming from `layout-loader` of Gatsby. I tried to follow the guidelines for dynamic page creation (as we do that a lot); I even tried to remove layouts altogether and create logic on `gatsby-browser.ts` that would encapsulate (wrap) the elements on their correct layouts, but no luck. The `children` prop is always empty.

So it's been quite a heptic journey as MDX v2 itself would already be a blocker, making compilation fail for hundreds of files because silly rules, that I understand that standardisation is critical, but sometimes it is an excellent ask that maintainers cannot do. And the `gatsby-plugin-mdx` that also isn't working correctly... And always giving strange errors.

I prefer to stay with the `v3` version of the plugin and MDX v1. But, because of the size of our Markdown files, Babel often says, `The code generator has deoptimised the styling since it exceeds the max of 500KB.`. Even when I manually modify the source of `gatsby-plugin-mdx` (changing the `mdx-loader`) to use Babel's `compact` mode, the whole output is still generated with whitespace. I have no idea why the code generation _needs_ to have so many blank lines and empty spaces; why isn't it minified (I would love it if y'all could update the v3 of the plugin to result in a minified JSX code to be passed towards `MDXRenderer`).

Well, this is my journey. Open to suggestions, but even if we make the plugin work correctly for our repository, there are still MDX v2 migration and Internationalisation problems...

Thanks!

**Reply by ovflowd** [reply]
Note.: Thank you @PaulieScanlon for that blog post, truly makes more sense. Yet, this basically means we need to migrate to MDX v2 syntax, and I do not see an easy way of doing so. I could just update our script by adding way more Regex rules to modify contents to match, but this would make transpilation of the source Markdown files from (nodejs/node) via the `npm run sync-api` take way longer, not to mention potential false positives.

We're talking about eventually bringing most of the modern Node.js API docs to the repository and also migrating all blog posts. We're talking about thousands of Markdown files. And I'm already exhausting my options of optimising the queries and infrastructure performance by, for example, this PR (https://github.com/nodejs/nodejs.dev/pull/2748)

**Reply by PaulieScanlon** [reply]
Hey @ovflowd would using the eslint approach work? If you corrected the issues in each of the files once, and then committed the changes you wouldn't need to run over all of the files each time you create a build.

These content changes are difficult and I feel your pain. I have migrated a few sites now and the content changes were always the hardest. But, once you fix the problem characters the rest of the migration should be less problematic.

**Reply by ovflowd** [reply]
> Hey @ovflowd would using the eslint approach work? If you corrected the issues in each of the files once, and then committed the changes you wouldn't need to run over all of the files each time you create a build.

Important to mention that the actual source of the API docs is `nodejs/node`, we could use ESLint API to manually run and lint/fix each file, but could you point me to the place where these ESLint rules/config/plugins are? I would need some testing. (This could prove useful even for MDX v1)

**Reply by PaulieScanlon** [reply]
You could try using eslint-mdx

There's a link to the two packages and how to use them on my blog post under the heading **eslint-mdx**

https://paulie.dev/posts/2022/08/mdx-2-breaking-changes-and-gatsby-plugin-mdx-v4/#eslint-mdx

**Reply by ovflowd** [reply]
Also, just a small typo on your post `npm install es-lint` isn't it `npm install eslint`?

**Reply by ovflowd** [reply]
After running the command, I get 4 thousand errors, such as:

```

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-http-server.en.md
  17:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-installation.en.md
  11:71  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-module-os.en.md
  22:33  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-output-to-cli.en.md
  64:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-setimmediate.en.md
  0:0  error  Parsing error: Expected value to be truthy

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-terminate-program.en.md
  28:43  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-writing-files.en.md
  71:36  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/package-json.en.md
  134:50  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)
```

It didn't fix any of the things I thought it should fix.

**Reply by ovflowd** [reply]
Okay, I was able to tweak some stuff, and reduced it to 38 errors. Whose are:

```
/Users/cwunder/GitHub/Personal/nodejs.dev/README.md
  61:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/privacy.en.md
  15:161  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/privacy.fr.md
  15:232  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/resources.en.md
  11:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/resources.fr.md
  11:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/security.en.md
  66:10  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/about/security.fr.md
  57:10  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/buffer.en.md
  886:18  error  Parsing error: Could not parse expression with acorn: Expecting Unicode escape sequence \uXXXX

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/deprecations.en.md
  896:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/dns.en.md
  289:23  error  Parsing error: Could not parse expression with acorn: Expecting Unicode escape sequence \uXXXX

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/errors.en.md
  69:4  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/esm.en.md
  765:64  error  Parsing error: Could not parse expression with acorn: Unexpected token

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/http2.en.md
  451:10  error  Parsing error: Could not parse expression with acorn: Unexpected content after expression

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/n-api.en.md
  2352:5  error  Parsing error: Unexpected character `=` (U+003D) before name, expected a character that can start a name, such as a letter, `$`, or `_`

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/os.en.md
  302:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/process.en.md
  3369:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/repl.en.md
  726:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/tls.en.md
  1434:30  error  Parsing error: Could not parse expression with acorn: Expecting Unicode escape sequence \uXXXX

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/util.en.md
  903:22  error  Parsing error: Could not parse expression with acorn: Expecting Unicode escape sequence \uXXXX

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/vm.en.md
  524:8  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/api/v18/zlib.en.md
  689:20  error  Parsing error: Could not parse expression with acorn: Unexpected content after expression

/Users/cwunder/GitHub/Personal/nodejs.dev/content/blog/announcements/2022-04-19-v18-release-announce.md
  11:54  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/blog/vulnerabilities/2022-01-11-jan-2022-security-releases.md
  105:60  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/blog/vulnerabilities/2022-03-18-mar-2022-security-releases.md
  87:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/blog/vulnerabilities/2022-05-05-openssl-fixes-in-regular-releases-may2022.md
  41:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/blog/vulnerabilities/2022-06-21-openssl-fixes-in-regular-releases-jun2022.md
  25:8  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/javascript-async-await.en.md
  57:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-event-emitter.en.md
  69:68  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-event-loop/index.en.md
  55:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-history.en.mdx
  19:3  error  Parsing error: Expected the closing tag `</a>` either after the end of `paragraph` (20:22) or another opening tag after the start of `paragraph` (18:5)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-http-server.en.md
  17:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-installation.en.md
  11:71  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-module-os.en.md
  22:33  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-output-to-cli.en.md
  64:2  error  Parsing error: Unexpected character `!` (U+0021) before name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a comment in MDX, use `{/* text */}`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-setimmediate.en.md
  0:0  error  Parsing error: Expected value to be truthy

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-terminate-program.en.md
  28:43  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/node-writing-files.en.md
  71:36  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)

/Users/cwunder/GitHub/Personal/nodejs.dev/content/learn/package-json.en.md
  134:50  error  Parsing error: Unexpected character `/` (U+002F) before local name, expected a character that can start a name, such as a letter, `$`, or `_` (note: to create a link in MDX, use `[text](url)`)
```

**Reply by ovflowd** [reply]
@PaulieScanlon I suppose there’s no way to instruct eslint to automatically fix those?

**Reply by PaulieScanlon** [reply]
Not that I know of. Is there a --fix arg you can pass? Making the changes
manually might be slower but you will at least be able to check the changes are correct before committing.

**Reply by PaulieScanlon** [reply]
Using the advanced find and place in VS Code will allow you to make changes in bulk, rather than one by one. Eslint is probably only for catching the errors, rather than fixing them, if you know what I mean.

**Reply by ovflowd** [reply]
I need to remind you that our docs are maintained in another repository that honestly couldn't care less about MDX/MD standards, we use our own thing. (Which is still compliant with general Markdown)

What I'm trying here is a best effort to make it be compatible with MDXv2 by running a script. It is truly bothersome that MDXv2 has such a restricting syntax that many other Markdown parsers, such as Remark, CommonMark and others are just fine with it.

Sorry, not gonna proceed with MDXv2, and I truly hope that the Gatsby team at least:

- Keep support to MDXv1 as many are using it
- Very nicely fix the memory leaks of `gatsby-plugin-mdx`v3 that cause Babel runtime code generator to scream. That'd be amazing, for real.

Thanks, and have a good one!

**Reply by PaulieScanlon** [reply]
Hey @ovflowd with regard to your other questions would you be able to start separate discussions. Whilst in your case they are all kind of related it might help us and others if you split them out into different discussions.

**Reply by ovflowd** [reply]
Sure thing 😄

**Comment by PaulieScanlon** [comment]
Hey @ovflowd i think this is the right move for you. If you don't want to, or can't fix the issues introduced by upgrading a major version of a package then it's probably best to stick with the older version.

**Reply by ovflowd** [reply]
>  If you don't want to,

It's not ideal to make such assessment when I'm actually trying to make the migration for the last days already. There still open questions that haven't been answered and they're blockers for our migration.

Don't get the wrong idea, I'd be more than happy to migrate to V2, **but**, the Node.js project is massive and enforcing such a big change on our documentation schema is **not** trivial.

Also, there are still open questions regarding Internationalisation and other things I asked that weren't answered, and it'd be great to be answered. 🙇
